Tests/js/views/dashboard/assets/ez-dashboardblockallcontentview-tests.js

```diff
@@ -0,0 +1,245 @@
+                contentTypeMock = new Y.Mock(),
+                modificationDate = new Date().toString(),
+                templateCalled = false,
+                origTpl = view.template,
+                container = view.get('container'),
+                prepareMock = function (mock, values, callCount) {
+                    Y.Mock.expect(mock, {
+                        method: 'get',
+                        args: [Y.Mock.Value.String],
+                        run: function (key) {
+                            if (values[key]) {
+                                return values[key];
+                            } else {
+                                Y.Assert.fail('Trying to get unexpected attribute value: ' + key);
+                            }
+                        },
+                        callCount: callCount
+                    });
+                };
+
+            prepareMock(contentMock, {
+                id: 'alter-bridge',
+                name: 'Alter Bridge',
+                resources: {MainLocation: '/location/1/2/3'},
+                mainLanguageCode: 'eng-GB',
+                currentVersion: versionMock
+            }, 5);
+
+            prepareMock(versionMock, {
+                versionNo: '2013',
+                modificationDate: modificationDate
+            }, 2);
+
+            prepareMock(contentTypeMock, {
+                names: {'eng-GB': 'Rock'}
+            }, 1);
+
+            view.template = function () {
+                templateCalled = true;
+
+                return origTpl.apply(this, arguments);
+            };
+
+            view.set('contentList', [{
+                content: contentMock,
+                contentType: contentTypeMock
+            }]);
+
+            Y.Assert.isTrue(templateCalled, 'The template should have been used to render view');
+            Y.Assert.areSame(1, container.one(SELECTOR_CONTENT).all('tr').size(), 'Should render 1 row');
+            Y.Assert.isFalse(container.hasClass(CLASS_LOADING), 'Should remove the loading state CSS class from the view container');
+
+            Y.Mock.verify(contentMock);
+            Y.Mock.verify(versionMock);
+            Y.Mock.verify(contentTypeMock);
+        }
+    });
+
+    getDataTest = new Y.Test.Case({
+        name: 'eZ Dashboard All Content Block View get data test',
+
+        setUp: function () {
+            this.view = new Y.eZ.DashboardBlockAllContentView(VIEW_CONFIG);
+        },
+
+        tearDown: function () {
+            this.view.destroy();
+        },
+
+        'Should get data when view get active': function () {
+            var view = this.view,
+                isEventFired = false;
+
+            view.on('getAllContent', function () {
+                isEventFired = true;
+            });
+
+            view.set('active', true);
+
+            Y.Assert.isTrue(isEventFired, 'Should fire the `getAllContent` event');
+        }
+    });
+
+    uiEventsTest = new Y.Test.Case({
```

Comment:
also I would add a test with 2 contents item and to check that a previously selected row is unselected when you select a new one.